### PR TITLE
fix bug #18

### PR DESCRIPTION
**Pull request by philoles**
Fixed this bug 

https://github.com/jolestar/go-commons-pool/issues/18

Also thanks for your great work.

### CODE REVIEW

**Comment by philoles** [comment]
I tested on my computer and all tests are passed. 
`
H:\repository\go\src\github.com\philoles\go-commons-pool>go test
RegistryEvictionPolicy github.com/jolestar/go-commons-pool/DefaultEvictionPolicy
0xc042037c80
RegistryEvictionPolicy github.com/jolestar/go-commons-pool/TestEvictionPolicy
PASS
ok      github.com/philoles/go-commons-pool     79.667s
`

**Comment by philoles** [comment]
Maybe package "context" was not add in go old version.

File: pool.go [hunk 1]
```diff
@@ -498,22 +501,60 @@ func (pool *ObjectPool) StartEvictor() {
...
+					return
+				case <-pool.evictor.C:
+					pool.evict()
+					pool.ensureMinIdle()
+				}
 			}
 		}()
 	}
+
+	// =========== bug code ==============
```
Comment (jolestar) [inline pool.go]: old code should be removed.

**Comment by jolestar** [comment]
@philoles please remove go 1.4, 1.6 support from .traivs.yaml

**Comment by philoles** [comment]
This is a memory leak bug. How do you test it? The only way I know to test this kind of bug is to run the code many many times and check if the memory getting bigger. 

Also this is a huge bug. It must be fixed otherwise this package can not be used. Also my solusion for this is not a grace one. I viewed the code and I think it needs a huge refactoring to fix this bug in a grace way. So I don't want to spent more time on it.

**Comment by jolestar** [comment]
@philoles I try to write a test case for that bug, and submit a pr https://github.com/jolestar/go-commons-pool/pull/21 . Please help to review, my solution is similar to yours,  but instead of context with chan, in order to be compatible with previous versions of go 1.7 .

**Comment by philoles** [comment]
I think your code will work. My solution is kind of complicated and yours is more complicated.  It should be simple and easy to be understood. But it works and fix the bug anyway.

**Pull request closed without merging**